pyensign/ensign.py

```diff
@@ -113,12 +113,11 @@ async def publish(self, topic_name, *events, client_id=""):
             topic_id = ULID.from_bytes(topic.id)
 
         # TODO: Support user-defined generators
-        def next():
-            for event in events:
-                yield event.proto()
+        event = next(event_source)
+        event_proto = event.proto()
```

Comment:
It looks like this is only consuming the first item in the generator right? Do you know if the original code works without modifications?